src/reflect/scala/reflect/internal/Reporting.scala

```diff
@@ -94,15 +94,18 @@ abstract class Reporter {
   class Severity(val id: Int)(name: String) { var count: Int = 0 ; override def toString = name}
   object INFO    extends Severity(0)("INFO")
   object WARNING extends Severity(1)("WARNING")
-  object ERROR   extends Severity(2)("ERROR")
+  private[this] var isERRORInitialized = false
+  object ERROR   extends Severity(2)("ERROR") {
+    isERRORInitialized = true
+  }
 
   def count(severity: Severity): Int       = severity.count
   def resetCount(severity: Severity): Unit = severity.count = 0
 
   def errorCount: Int   = count(ERROR)
```

Comment:
Asking for errorCount would init ERROR. Maybe set the flag from error and reset from reset? I think that's the intention in the filtering reporter, namely, we have an error when an ERROR is actually passed, not just when it's issued. "The ERROR has landed."

But setting the flag from the initializer is pretty clever. I'll try to refresh my memory about reporters.

Reply:
Looks like `errorCount` is only called under `-Xmaxerrs` / `-Xmaxwarns`, which isn't exercised under my profile setup. I might as well add the check for `isERRORInitialized`.